framework/src/mesh/MooseMesh.C

```diff
@@ -1569,6 +1571,19 @@ MooseMesh::detectPairedSidesets()
     }
   }
 
+  // For a distributed mesh, boundaries may be distributed as well. We therefore
+  // collect information from everyone.
+  if (_use_distributed_mesh)
+    for (unsigned side_dim = 0; side_dim < dim; ++side_dim)
+    {
+      _communicator.set_union(minus_x_ids[side_dim]);
+      _communicator.set_union(plus_x_ids[side_dim]);
+      _communicator.set_union(minus_y_ids[side_dim]);
+      _communicator.set_union(plus_y_ids[side_dim]);
+      _communicator.set_union(minus_z_ids[side_dim]);
+      _communicator.set_union(plus_z_ids[side_dim]);
```

Comment:
You're the parallel guy so I'll defer to you on this, but is 6 communications in a row ideal? I'm wondering if there's a way to do this all in one go, but if you don't expect this to be important than I'm not going to worry about it.

Reply:
Actually more than 6. There are 6*dim communications. I did not find an API that can do all these using one global communication. 

@roystgnr Any API? I want to do  6*dim communications all at once.

Reply:
I'm afraid we don't have an API to do that without repacking into a different data structure first.  In this case a vector of tuples of 6 boundary_id_type would work best if your sets are nearly the same size, or a vector of ```pair<boundary_id_type, int>``` would work better (where the int is an index from 0 to 5 identifying which set each came from) if the vector of tuples would be too sparse.

The only other alternative is something I've been thinking about for a while - adding optional Request arguments to a lot more APIs, so you can interleave computation and communication even on things like gathers and reductions.  That'll be more long term, but if you're interested then I'll bump it up from "idle speculation" to "thing I actually intend to do eventually".  ;-)